app/Repositories/GridRepository.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Repositories;
+
+use Contracts\Repositories\GridRepositoryContract;
+use Illuminate\Cache\Repository;
+use Waynestate\Api\Connector;
+use Waynestate\Promotions\ParsePromos;
+
+class GridRepository implements GridRepositoryContract
+{
+    /** @var Connector */
+    protected $wsuApi;
+
+    /** @var ParsePromos */
+    protected $parsePromos;
+
+    /** @var Repository */
+    protected $cache;
+
+    /**
+     * Construct the repository.
+     *
+     * @param Connector $wsuApi
+     * @param ParsePromos $parsePromos
+     * @param Repository $cache
+     */
+    public function __construct(Connector $wsuApi, ParsePromos $parsePromos, Repository $cache)
+    {
+        $this->wsuApi = $wsuApi;
+        $this->parsePromos = $parsePromos;
+        $this->cache = $cache;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getGridPromos()
+    {
+        // If there is an grid custom page field then inject it into the group reference
+        if (!empty($data['data']['grid_promo_group_id']) && ! array_key_exists($data['data']['grid_promo_group_id'], $group_reference)) {
+            $group_reference[$data['data']['grid_promo_group_id']] = 'grid_promos';
+        }
+
+        $group_reference = [];
+
+        $group_config = [
+            'grid_promos' => 'limit:75',
```

Comment:
Thoughts on this being configurable in the function call? 

```
public function getGridPromos($limit = 75)
{
...

$group_config = [
  'grid_promos' => 'limit:' . $limit,

...
```

Reply:
Ok, added this.